contracts/src/SuccinctStaking.sol

```diff
@@ -289,16 +290,13 @@ contract SuccinctStaking is
         if (unstakeClaims[msg.sender].length >= maxUnstakeRequests) revert TooManyUnstakeRequests();
 
         // Check that this prover is not in the process of being slashed.
-        if (slashClaimCount[prover] > 0) revert ProverHasSlashRequest();
+        _requireProverWithoutSlashRequests(prover);
 
         // Get the amount of $stPROVE this staker currently has.
         uint256 stPROVEBalance = balanceOf(msg.sender);
 
-        // Get the amount $stPROVE this staker has in pending unstake claims.
-        uint256 stPROVEClaim = _getUnstakeClaimBalance(msg.sender);
-
         // Check that this staker has enough $stPROVE to unstake this amount.
-        if (stPROVEBalance < stPROVEClaim + _stPROVE) revert InsufficientStakeBalance();
+        if (stPROVEBalance < _stPROVE) revert InsufficientStakeBalance();
```

Comment:
I think you don't actually need this check because the burn will fail if it's not true?

Reply:
its from the other pr